pinot-core/src/test/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManagerTest.java

```diff
@@ -760,6 +780,9 @@ protected boolean buildSegmentAndReplace() {
     @Override
     protected String buildSegmentInternal(boolean forCommit) {
       _buildSegmentCalled = true;
+      if (_failSegmentBuild) {
```

Comment:
Seems like this test is very heavily coupled to the actual implementation (white box) as evidenced by the numerous flags (eg. `_buildAndReplaceCalled`). Do we have plans to refactor this test into something cleaner at some point?